etl/steps/data/garden/wb/2023-04-30/income_groups.py

```diff
@@ -0,0 +1,123 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+import numpy as np
+import pandas as pd
+from owid.catalog import Dataset, Table, VariableMeta
+from structlog import get_logger
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+EXPECTED_MISSING_COUNTRIES_IN_LATEST_RELEASE = {
+    "Czechoslovakia",
+    "Mayotte",
+    "Netherlands Antilles",
+    "Serbia and Montenegro",
+    "USSR",
+    "Venezuela",
+    "Yugoslavia",
+}
+
+
+def run(dest_dir: str) -> None:
+    log.info("income_groups: start")
+
+    #
+    # Load inputs.
+    #
+    # Load meadow dataset.
+    ds_meadow: Dataset = paths.load_dependency("income_groups")
+
+    # Read table from meadow dataset.
+    tb_meadow = ds_meadow["income_groups"]
+
+    # Create a dataframe with data from the table.
+    df = pd.DataFrame(tb_meadow)
+
+    #
+    # Process data for main table.
+    #
+    log.info("income_groups: harmonize_countries")
+    df = geo.harmonize_countries(df=df, countries_file=paths.country_mapping_path)
+    # Harmonize income group labels
+    log.info("income_groups: harmonize income group labels")
+    df = harmonize_income_group_labels(df)
+    # Set index, drop code column
+    df = df.drop(columns=["country_code"]).set_index(["country", "year"], verify_integrity=True).sort_index()
+
+    #
+    # Process data for table with values of latest release
+    #
+    # Get classifications for each country for the latest year available.
+    df_latest = df.reset_index().drop_duplicates(subset=["country"], keep="last")
+    # Sanity checks of missing countries for latest release
+    missing_countries = set(df_latest.loc[df_latest["year"] != df_latest["year"].max(), "country"])
+    assert (
+        missing_countries == EXPECTED_MISSING_COUNTRIES_IN_LATEST_RELEASE
+    ), f"Unexpected missing countries in latest release. All missing countries: {missing_countries}"
+    # Get data only for latest release (and remove column year)
+    df_latest = df_latest[df_latest["year"] == df_latest["year"].max()].drop(columns=["year"])
```

Comment:
Indeed. I sanity check that missing countries are as expected.

The way I thought about this table table is that it should only contain countries as per the latest release (2022). So, the classification for Venezuela is none, as expected. 

But ofc, if you think it makes more sense to have data for whatever the latest year available, that's easy to change

Reply:
I'm not sure if this has important consequences. On the one hand, I think all countries should have their "latest" income group classification (including, say, USSR). On the other hand, if WB decided to not assign any income group to a certain country (e.g. Venezuela) because of recent changes and lack of a meaningful classification, then it should be nan.

Reply:
OK, given that the only affected countries are those in `EXPECTED_MISSING_COUNTRIES_IN_LATEST_RELEASE`, maybe it's a good criterion to only take the latest year (I think that would be what we currently have, since we don't have income group for, say, USSR). If more countries are affected in a future release, we can rethink it.